### PR TITLE
*: Add antiaffinity to Prometheus & Alertmanager

### PR DESCRIPTION
**Pull request by squat**
this is a manual patch of https://github.com/openshift/cluster-monitoring-operator/pull/117 for 3.11

cc @mxinden @metalmatze

### CODE REVIEW

File: jsonnet/main.jsonnet [hunk 1]
```diff
@@ -1,4 +1,5 @@
 local kp = (import 'kube-prometheus/kube-prometheus.libsonnet') +
+           (import 'anti-affinity/kube-prometheus-anti-affinity.libsonnet') +
```
Comment (mxinden) [inline jsonnet/main.jsonnet]: Can you add a comment here why we are doing it?

**Comment by mxinden** [comment]
Could you add some details to the commit message why we are doing the hack via the second jsonnetfile entry?

**Comment by squat** [comment]
updated the commit message and added a note in the main.jsonnet

**Comment by brancz** [comment]
I'm not sure how significant this is but this changes our requirement of 1 infra node to at least 3 infra nodes (meaning things like minishift will never work entirely correct). I'm expecting this to cause bugs to be opened, but I do prefer having this. So what do you say we make the replicas of alertmanager/prometheus configurable when a bug does get opened about this? Then for those types of clusters we can recommend setting the replicas to 1 for Prometheus and Alertmanager.

**Pull request merged**